src/models/application/preferences/mod.rs

```diff
@@ -434,6 +457,42 @@ mod tests {
         assert_eq!(preferences.soft_tabs(Some(PathBuf::from("Makefile")).as_ref()), false);
     }
 
+    #[test]
+    fn syntax_definition_name_returns_user_defined_syntax_by_extension_for_full_filename() {
+        let data = YamlLoader::load_from_str("types:\n  xyz:\n    syntax: Rust").unwrap();
+        let preferences = Preferences::new(data.into_iter().nth(0));
+
+        assert_eq!(preferences.syntax_definition_name(&Path::new("test.xyz")),
+                   Some("Rust".to_owned()));
+    }
+
+    #[test]
+    fn syntax_definition_name_returns_user_defined_syntax_by_extension_for_deep_filename() {
+        let data = YamlLoader::load_from_str("types:\n  xyz:\n    syntax: Rust").unwrap();
+        let preferences = Preferences::new(data.into_iter().nth(0));
+
+        assert_eq!(preferences.syntax_definition_name(&Path::new("src/test.xyz")),
+                   Some("Rust".to_owned()));
+    }
+
+    #[test]
+    fn syntax_definition_name_returns_user_defined_syntax_for_full_filename_without_extension() {
+        let data = YamlLoader::load_from_str("types:\n  Makefile:\n    syntax: Makefile").unwrap();
+        let preferences = Preferences::new(data.into_iter().nth(0));
+
+        assert_eq!(preferences.syntax_definition_name(&Path::new("Makefile")),
+                   Some("Makefile".to_owned()));
+    }
+
```

Comment:
I was thinking it would be good to test that we support full filename matching on nested paths, too:

```suggestion

    #[test]
    fn syntax_definition_name_returns_user_defined_syntax_for_full_deep_filename() {
        let data = YamlLoader::load_from_str("types:\n  Makefile:\n    syntax: Makefile").unwrap();
        let preferences = Preferences::new(data.into_iter().nth(0));

        assert_eq!(preferences.syntax_definition_name(&Path::new("src/Makefile")),
                   Some("Makefile".to_owned()));
    }

```